File: Backend/plated/src/main/java/com/priyanshparekh/plated/recipe/dto/AddRecipeRequest.java
package com.priyanshparekh.plated.recipe.dto;

import com.priyanshparekh.plated.ingredient.Ingredient;
import com.priyanshparekh.plated.ingredient.dto.IngredientDto;
import com.priyanshparekh.plated.recipe.recipeingredient.dto.RecipeIngredientDto;
import com.priyanshparekh.plated.recipe.step.StepsDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class AddRecipeRequest {

    private String recipeName;
    private String description;
    private int servingSize;
    private float cookingTime;
    private String cuisine;
    private String category;
    private List<RecipeIngredientDto> ingredientList;
    private List<StepsDto> stepList;

}

File: Frontend/Plated2/core-model/src/main/java/com/priyanshparekh/core/model/recipe/AddRecipeRequest.kt
package com.priyanshparekh.core.model.recipe

data class AddRecipeRequest(
    val userId: Long,
    var recipeName: String,
    val description: String,
    val servingSize: Int,
    val cookingTime: Long,
    val cuisine: String,
    val category: String,
    val ingredientList: List<RecipeIngredientDto>,
    val stepList: List<StepDto>
) {
    constructor(): this(-1L, "", "", 0, 0L, "", "", listOf(), listOf())
}
